### PR TITLE
refactor work group scan and reduction tests

### PR DESCRIPTION
**Pull request by bashbaug**
This PR refactors the work-group scan and reduction tests to reduce duplication and make it easier to add support for other scan and reduction operators or types in the future.

### CODE REVIEW

File: test_conformance/workgroups/test_wg_scan_reduce.cpp
Comment (gwawiork) [inline test_conformance/workgroups/test_wg_scan_reduce.cpp:140]: please exchange variable name to more friendly instead of `m`

File: test_conformance/workgroups/test_wg_scan_reduce.cpp
Comment (gwawiork) [inline test_conformance/workgroups/test_wg_scan_reduce.cpp:131]: It would be nice to have in `testName` the `testOpName` .  Printing output would be more readable than now:
work_group_scan_inclusive_add...
work_group_scan_inclusive int passed
work_group_scan_inclusive uint passed
work_group_scan_inclusive long passed
work_group_scan_inclusive ulong passed
work_group_scan_inclusive_add passed
work_group_scan_inclusive_min...
work_group_scan_inclusive int passed
work_group_scan_inclusive uint passed
work_group_scan_inclusive long passed
work_group_scan_inclusive ulong passed
work_group_scan_inclusive_min passed
work_group_scan_inclusive_max...
work_group_scan_inclusive int passed
work_group_scan_inclusive uint passed
work_group_scan_inclusive long passed
work_group_scan_inclusive ulong passed
work_group_scan_inclusive_max passed
work_group_scan_exclusive_add...
work_group_scan_exclusive int passed
work_group_scan_exclusive uint passed
work_group_scan_exclusive long passed
work_group_scan_exclusive ulong passed
work_group_scan_exclusive_add passed
work_group_scan_exclusive_min...
work_group_scan_exclusive int passed
work_group_scan_exclusive uint passed
work_group_scan_exclusive long passed
work_group_scan_exclusive ulong passed
work_group_scan_exclusive_min passed
work_group_scan_exclusive_max...
work_group_scan_exclusive int passed
work_group_scan_exclusive uint passed
work_group_scan_exclusive long passed
work_group_scan_exclusive ulong passed
work_group_scan_exclusive_max passed

File: test_conformance/workgroups/test_wg_scan_reduce.cpp
Comment (gwawiork) [inline test_conformance/workgroups/test_wg_scan_reduce.cpp:30]: tid should be `size_t` type

File: test_conformance/workgroups/test_wg_scan_reduce.cpp
Comment (gwawiork) [inline test_conformance/workgroups/test_wg_scan_reduce.cpp:37]: tid should be `size_t` type

**Comment by gwawiork** [comment]
Generally looks very good. Not "must have" remarks in comments.

File: test_conformance/workgroups/test_wg_scan_reduce.cpp
Comment (bashbaug) [reply test_conformance/workgroups/test_wg_scan_reduce.cpp]: Fixed - thanks!  Please take a look.

File: test_conformance/workgroups/test_wg_scan_reduce.cpp
Comment (bashbaug) [reply test_conformance/workgroups/test_wg_scan_reduce.cpp]: Fixed - thanks! Please take a look.

File: test_conformance/workgroups/test_wg_scan_reduce.cpp
Comment (bashbaug) [reply test_conformance/workgroups/test_wg_scan_reduce.cpp:131]: Done - nice catch!  I thought I had the output exactly the same but I missed this one.

File: test_conformance/workgroups/test_wg_scan_reduce.cpp
Comment (bashbaug) [reply test_conformance/workgroups/test_wg_scan_reduce.cpp:30]: I'd prefer to change this later, if needed.  My goal with this PR was to generate more-or-less exactly the same test content, just in a more efficient way.  I agree that a size_t is "more right", but in this case we know our maximum ND-range size will be less than 4B work-items so we should be OK.

File: test_conformance/workgroups/test_wg_scan_reduce.cpp
Comment (bashbaug) [reply test_conformance/workgroups/test_wg_scan_reduce.cpp:37]: Same comment as above.  I'd prefer to change this later, if needed.  My goal with this PR was to generate more-or-less exactly the same test content, just in a more efficient way.  I agree that a size_t is "more right", but in this case we know our maximum ND-range size will be less than 4B work-items so we should be OK.

**Review by gwawiork — APPROVED** [review]
Thanks! LGTM

**Review by Jeremy-Kemp — APPROVED** [review]
Great improvement, thanks!

LGTM, works for us.

**Comment by bashbaug** [comment]
This has two approvals and a +1 from Stuart a few weeks ago, so merging to unblock future improvements.

**Pull request merged**
